### PR TITLE
Fix issue where sometimes Titan strategy is not applied

### PR DESCRIPTION
**Pull request by aelred**
Needs 5 passes

### CODE REVIEW

**Comment by aelred** [comment]
!rtg - one pass

**Comment by aelred** [comment]
!rtg - `TaskManagerTest.whenRecurringTaskSubmitted_ItExecutesMoreThanOnce`

**Comment by aelred** [comment]
!rtg - failed because of some sort of time-out?

**Comment by aelred** [comment]
!rtg - two passes!

**Comment by aelred** [comment]
!rtg - two passes!

**Comment by aelred** [comment]
!rtg - looks like jenkins just didn't run?

**Comment by aelred** [comment]
!rtg - three passes!

**Comment by aelred** [comment]
!rtg - four passes!

**Comment by aelred** [comment]
five passes 🎉

**Comment by aelred** [comment]
!rtg - failure on `CommitLogControllerTest.whenCommittingGraph_TaskManagerReceivesCountTask`

File: grakn-factory/titan-factory/src/main/java/ai/grakn/factory/TitanPreviousPropertyStepStrategy.java [hunk 1]
```diff
@@ -56,9 +56,9 @@
     public void apply(Traversal.Admin<?, ?> traversal) {
 
         // Retrieve all graph (`V()`) steps - this is the step the strategy should replace
-        List<TitanGraphStep> graphSteps = TraversalHelper.getStepsOfClass(TitanGraphStep.class, traversal);
+        List<GraphStep> graphSteps = TraversalHelper.getStepsOfAssignableClass(GraphStep.class, traversal);
```
Comment (fppt) [inline grakn-factory/titan-factory/src/main/java/ai/grakn/factory/TitanPreviousPropertyStepStrategy.java]: This kind of makes sense. Might have been ignoring the non Titan steps. I am still not sure why this led to the random failures.

**Comment by fppt** [comment]
@aelred please create a merge into master from stable and reference it here.

**Comment by aelred** [comment]
Merge into master here: #1734

**Pull request merged**